File: problems/starterpackages/TSP-Starter.cpp
#include <vector>
#include <iostream>
#include <fstream>
#include <string>
#include <time.h>

struct Point
{
	int x, y, z, index;
};

//Gets the problem as a vector of points from a specified filename.
std::vector<Point> getProblem(const std::string & filename)
{
	std::vector<Point> pts;
	pts.resize(500); //Number of points
	std::ifstream f;
	f.open(filename);
	for(int a = 0; a < 500; a++) f >> pts[a].index >> pts[a].x >> pts[a].y >> pts[a].z;
	f.close();
}

void outputSolutionsToFile(const std::string & name, const std::vector<int> & s)
{
	std::ofstream output(name + '-' + std::to_string(time(NULL)) + ".txt");
	if(output.is_open()) for(auto a = s.begin(); a != s.end(); a++) output << *a << ' ';
	else throw std::runtime_error("Couldn't open file");
}

int main()
{
	std::vector<Point> problem = getProblem("tsp.txt");
	
	std::vector<int> solution;
	
	//Put your code here to create the solution.

	outputSolutionsToFile("YOUR-NAME-HERE", solution);

	system("PAUSE");
	return EXIT_SUCCESS;
}

File: problems/scripts/TSP-Starter.py
import time

class Point:
	def __init__(self, x, y, z, index):
		self.x = x
		self.y = y
		self.z = z
		self.index = index

def getProblem(f):
    problem = []
    with open(f, 'r') as file:
        for line in file:
            components = line.split(' ')
            problem.append(Point(int(components[1]), int(components[2]), int(components[3]), int(components(0))))
    return problem

def outputSolutionToFile(name, solution):
	content = name+str(time.time())+".txt"
	for point in solution:
	open("output.txt", "w").write(content)

problem = getProblem("PUT FILENAME HERE")

solution = []

#Put your code here and add your path to the solutions list.
#It should be just a list of integers.

outputSolutionsToFile("PUT YOUR NAME HERE", solutions)